release/models/route-programming/openconfig-route-programming-state.yang

```diff
@@ -0,0 +1,184 @@
+    www.openconfig.net";
+
+  description
+    "This module provides detailed information as to the programming
+    state of routes within a particular network instance. It can be used to
+    track where there are routing programming errors on a device.";
+
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2021-10-16" {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  grouping route-state-top {
+    description
+      "Top-level grouping for route state in OpenConfig.";
+
+    container route-programming {
+      config false;
+      description
+        "The programming status of routes within the network-instance. Route
+        programming reports are divided into a set of sub-categories:
+
+         - failed programming -- these routes are those that were attempted to be
+           programed to the hardware, but the programming operation failed.
+         - stale programming -- these routes are those that are programmed into
+           hardware, but some failure occurred in updating to a new state.
+         - drop programming - these routes are explicitly programmed into hardware
+           to point to a destination that discards packets.";
+
+      container failed-routes {
+        description
+          "Surrounding container for the list of routes that fail programming.";
+
+        list failed {
+          key "prefix";
+
+          description
+            "A prefix that was attempted to the programmed into hardware, but the
+            programming operation failed.";
+
+          leaf prefix {
+            type leafref {
+              path "../state/prefix";
+            }
+            description
+              "Reference to the prefix that keys the failed list.";
+          }
+
+          container state {
+            description
+              "Operational state parameters relating to a failed programming
+              operation.";
+            uses route-state-common;
+          }
+        }
+      }
+
+      container stale-routes {
+        description
+          "Surrounding container for the list of routes that are currently in
+          a stale state.";
+
+        list stale {
+          key "prefix";
+
+          description
+            "A prefix that is currently installed in hardware, but a subsequent
+            operation to update its programming failed - such that the entry in
+            hardware is stale.";
+
+          leaf prefix {
+            type leafref {
+              path "../state/prefix";
+            }
+            description
+              "Reference to the prefix that keys the stale list.";
+          }
+
+          container state {
+            description
+              "Operational state parameters relating to a stale route.";
+            uses route-state-common;
+          }
+        }
+      }
+
+      container drop-routes {
```

Comment:
Thanks for this good comment.  I believe the original intent of this container was to represent errors.  Can you describe the use case(s) for an intentionally dropped route?  I ask because maybe there is another list that should be added for intentionally dropped routes.

Reply:
The most common use case would where we configure static null routes where we want to deliberately drop the traffic rather than having it get routed through the default/some less specific route.

Reply:
Yes that sounds about right. That use will get captured by the collection though as it is defined now. ( Since the packets for just prefixes will get dropped in the data plane )

Reply:
Ok, counting routes/RIB entries which are created and forwarding to a next hop such as null should be counted as successful routes programmed and added to the drop-routes/prefix list.  

If a 'null route' cannot be programmed, then it should be added to the failed-routes/prefix list.

Will update the description accordingly